CONTRIBUTING.md

```diff
@@ -17,26 +84,222 @@ augroup gno_autocmd
 augroup END
 ```
 
-TODO: other vim tweaks to make work with vim-go etc.
+#### Emacs Support
 
-### Getting started with Emacs
-
-Install [go-mode.el](https://github.com/dominikh/go-mode.el).
-
-Add to your emacs configuration file:
+1. Install [go-mode.el](https://github.com/dominikh/go-mode.el).
+2. Add to your emacs configuration file:
 
 ```lisp
 (add-to-list 'auto-mode-alist '("\\.gno\\'" . go-mode))
 ```
 
-## Notable Contributions
+### Local Setup
+
+To get started with Gno development, the process is relatively straightforward.
+
+Clone the repo:
+`git clone https://github.com/gnolang/gno.git`
+
+Build / install base commands:
+`make build `
+
+That’s it!
+
+### Testing
+
+There are essentially 2 ways to execute the entire test suite:
+
+1. Using the base `go test` commands (running locally),
+2. Using a tool like [act](https://github.com/nektos/act) to run workflow files in Docker containers
+
+#### Running locally
+
+To run the entire test suite locally, run the following command:
+`make test`
+
+This will execute the full test suite, that includes tests for `.gno` files, as well as project `.go` tests.
+
+#### Running test workflows
+
+Using a tool like [act](https://github.com/nektos/act) can enable you to run specific repository workflow files locally,
+using Docker. The workflow configurations contain different `go` versions, so it might be worth running if you’re
+worried about compatibility.
+
+To run the entire test suite through workflow files, run the following command:
+`act -v -j go-test`
+
+### Repository Structure
+
+The repository structure can seem tricky at first, but it’s simple if you consider the philosophy that the gno project
+employs (check out [PHILOSOPHY.md](https://github.com/gnolang/gno/blob/master/PHILOSOPHY.md)).
+
+The gno project currently favors a mono-repo structure, as it’s easier to manage contributions and keep everyone
+aligned. In the future, this may change, but in the meantime the majority of gno resources and source code will be
+centralized here.
+
+- `cmd` - contains the base command implementations for tools like `gnokey`, `gnotxport`, etc. The actual underlying
+  logic is located within the `pkgs` subdirectories.
+- `examples` - contains the example `.gno` realms and packages. This is the central point for adding user-defined realms
+  and packages.
+- `gnoland` - contains the base source code for bootstrapping the Gnoland node
+- `pkgs` - contains the dev-audited packages used throughout the gno codebase
+- `stdlibs` - contains the standard library packages used (imported) in `.gno` Smart Contracts. These packages are
+  themselves `.gno` files.
+- `tests` - contains the standard language tests for Gnolang
+
+## How do I?
+
+### How do I submit changes?
+
```

Comment:
In this section, I'd like something to discourage people from opening issues that are just the copy-paste of a PR.

Instead, we should consider PRs as issues with attached code and encourage people to use PRs for discussions while keeping issues for things that are not linked to a PR or multiple PRs.

Reply:
I've updated this section, as well as the section for `How do I request a feature?` to include a note on favoring PRs instead of issues:

[508279d](https://github.com/gnolang/gno/pull/469/commits/508279df401f3050a9e64a01da8ac9492394cfd5)

Reply:
👍